### PR TITLE
fix(pypi): normalize simple package lookup

### LINKED ISSUE #12682: pipenv aws-cdk dependencies cannot be resolved
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

29.8.4

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.4.2-ee

### Describe the bug

I have private Gitlab repository that contains a `Pipfile` with the following contents:

```
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]
"aws-cdk.aws-apigateway" = "==1.129.0"
"aws-cdk.aws-ec2" = "==1.129.0"
"aws-cdk.aws-ecs" = "==1.129.0"
"aws-cdk.aws-ecs-patterns" = "==1.129.0"
"aws-cdk.aws-iam" = "==1.129.0"
"aws-cdk.aws-lambda" = "==1.129.0"
"aws-cdk.aws-route53" = "==1.129.0"
"aws-cdk.aws-s3" = "==1.129.0"
"aws-cdk.aws-s3-assets" = "==1.129.0"
"aws-cdk.aws-secretsmanager" = "==1.129.0"
"aws-cdk.aws-ssm" = "==1.129.0"
"aws-cdk.core" = "==1.129.0"

[requires]
python_version = "3.10"
```

The renovate configuration is as follows:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":rebaseStalePrs",
    ":dependencyDashboard",
    "group:all"
  ],
  "prConcurrentLimit": 1
}
```

When I run (a dockerized instance of) renovate, I expect that it creates a MR in which all Pipenv dependencies are updated to their latest versions.
However, renovate reports dependency lookup warnings for all AWS CDK dependencies (see logs).

I tried to:
* Change the Python version in the Pipfile (3.8, 3.9);
* Change the dependency name (e.g. `aws-cdk-aws-apigateway = "==1.129.0"`);

but the results were the same.

The commands to run renovate is as follows:
```
docker run \
    -e LOG_LEVEL=DEBUG \
    -e RENOVATE_ENDPOINT=https://gitlab.com/api/v4 \
    -e RENOVATE_AUTODISCOVER=true \
    -e RENOVATE_TOKEN=<project_access_token> \
    -e RENOVATE_PLATFORM=gitlab \
    renovate/renovate:29.8.4
```

where `<project_access_token>` is a project access token with `api` scope.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /usr/src/app/config
DEBUG: No config file found on disk - skipping
DEBUG: File config
       "config": {}
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [],
         "platform": "gitlab",
         "endpoint": "[REDACTED]/api/v4",
         "token": "***********",
         "autodiscover": true
       }
DEBUG: Combined config
       "config": {
         "hostRules": [],
         "platform": "gitlab",
         "endpoint": "[REDACTED]/api/v4",
         "token": "***********",
         "autodiscover": true
       }
DEBUG: Adding trailing slash to endpoint
DEBUG: GitLab version is: 14.4.2-ee
DEBUG: Using platform gitAuthor: renovate <project468_bot@example.com>
DEBUG: Adding token authentication for [REDACTED] to hostRules
DEBUG: Using baseDir: /tmp/renovate
DEBUG: Using cacheDir: /tmp/renovate/cache
DEBUG: Initializing Renovate internal cache into /tmp/renovate/cache/renovate/renovate-cache-v1
DEBUG: Commits limit = null
DEBUG: Autodiscovering GitLab repositories
DEBUG: Discovered 1 project(s)
 INFO: Autodiscovered repositories
       "length": 1,
       "repositories": ["mn-dnp/playground/renovate"]
DEBUG: Clearing hostRules
DEBUG: Adding token authentication for [REDACTED] to hostRules
 INFO: Repository started (repository=mn-dnp/playground/renovate)
       "renovateVersion": "29.8.4"
DEBUG: Using localDir: /tmp/renovate/repos/gitlab/mn-dnp/playground/renovate (repository=mn-dnp/playground/renovate)
DEBUG: mn-dnp/playground/renovate default branch = main (repository=mn-dnp/playground/renovate)
DEBUG: Enabling Git FS (repository=mn-dnp/playground/renovate)
DEBUG: using http URL (repository=mn-dnp/playground/renovate)
       "url": "[REDACTED]/mn-dnp/playground/renovate.git"
DEBUG: resetMemCache() (repository=mn-dnp/playground/renovate)
DEBUG: Resetting npmrc (repository=mn-dnp/playground/renovate)
DEBUG: detectSemanticCommits() (repository=mn-dnp/playground/renovate)
DEBUG: Initializing git repository into /tmp/renovate/repos/gitlab/mn-dnp/playground/renovate (repository=mn-dnp/playground/renovate)
DEBUG: Performing blobless clone (repository=mn-dnp/playground/renovate)
DEBUG: git clone completed (repository=mn-dnp/playground/renovate)
       "durationMs": 1012
DEBUG: latest repository commit (repository=mn-dnp/playground/renovate)
       "latestCommit": {
         "hash": "9bc2513719b428fc4c54332b67094481bce7835c",
         "date": "2021-11-16T09:45:34+01:00",
         "message": "make dependencies quoted",
         "refs": "HEAD -> main, origin/main, origin/HEAD",
         "body": "",
         "author_name": "Tom",
         "author_email": "tom@test.nl"
       }
DEBUG: getCommitMessages (repository=mn-dnp/playground/renovate)
DEBUG: Semantic commits detection: unknown (repository=mn-dnp/playground/renovate)
DEBUG: No semantic commits detected (repository=mn-dnp/playground/renovate)
DEBUG: checkOnboarding() (repository=mn-dnp/playground/renovate)
DEBUG: isOnboarded() (repository=mn-dnp/playground/renovate)
DEBUG: findFile(renovate.json) (repository=mn-dnp/playground/renovate)
DEBUG: Config file exists (repository=mn-dnp/playground/renovate)
       "fileName": "renovate.json"
DEBUG: ensureIssueClosing() (repository=mn-dnp/playground/renovate)
DEBUG: Repo is onboarded (repository=mn-dnp/playground/renovate)
DEBUG: Found renovate.json config file (repository=mn-dnp/playground/renovate)
DEBUG: Repository config (repository=mn-dnp/playground/renovate)
       "fileName": "renovate.json",
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "config:base",
           ":rebaseStalePrs",
           ":dependencyDashboard",
           "group:all"
         ],
         "baseBranches": ["main"],
         "prConcurrentLimit": 5
       }
DEBUG: migrateAndValidate() (repository=mn-dnp/playground/renovate)
DEBUG: No config migration necessary (repository=mn-dnp/playground/renovate)
DEBUG: massaged config (repository=mn-dnp/playground/renovate)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "config:base",
           ":rebaseStalePrs",
           ":dependencyDashboard",
           "group:all"
         ],
         "baseBranches": ["main"],
         "prConcurrentLimit": 5
       }
DEBUG: migrated config (repository=mn-dnp/playground/renovate)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": [
           "config:base",
           ":rebaseStalePrs",
           ":dependencyDashboard",
           "group:all"
         ],
         "baseBranches": ["main"],
         "prConcurrentLimit": 5
       }
DEBUG: Setting hostRules from config (repository=mn-dnp/playground/renovate)
DEBUG: Found repo ignorePaths (repository=mn-dnp/playground/renovate)
       "ignorePaths": [
         "**/node_modules/**",
         "**/bower_components/**",
         "**/vendor/**",
         "**/examples/**",
         "**/__tests__/**",
         "**/test/**",
         "**/tests/**",
         "**/__fixtures__/**"
       ]
DEBUG: No vulnerability alerts found (repository=mn-dnp/playground/renovate)
DEBUG: findIssue(Dependency Dashboard) (repository=mn-dnp/playground/renovate)
DEBUG: baseBranches (repository=mn-dnp/playground/renovate)
       "baseBranches": ["main"]
DEBUG: baseBranch: main (repository=mn-dnp/playground/renovate)
DEBUG: extract() (repository=mn-dnp/playground/renovate)
DEBUG: Setting current branch to main (repository=mn-dnp/playground/renovate)
DEBUG: latest commit (repository=mn-dnp/playground/renovate)
       "branchName": "main",
       "latestCommitDate": "2021-11-16T09:45:34+01:00"
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager ansible-galaxy (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)galaxy\.ya?ml$ for manager ansible-galaxy (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: azure.*pipelines?.*\.ya?ml$ for manager azure-pipelines (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)batect(-bundle)?\.yml$ for manager batect (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)batect$ for manager batect-wrapper (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)WORKSPACE(|\.bazel)$ for manager bazel (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.bzl$ for manager bazel (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.?bitbucket-pipelines\.ya?ml$ for manager bitbucket-pipelines (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.cake$ for manager cake (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Cargo.toml$ for manager cargo (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/).circleci/config.yml$ for manager circleci (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)cloudbuild.ya?ml for manager cloudbuild (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)([\w-]*)composer.json$ for manager composer (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)deps\.edn$ for manager deps-edn (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)docker-compose[^/]*\.ya?ml$ for manager docker-compose (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/|\.)Dockerfile$ for manager dockerfile (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Dockerfile\.[^/]*$ for manager dockerfile (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/).drone.yml$ for manager droneci (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/).gitmodules$ for manager git-submodules (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: ^(workflow-templates|\.github\/workflows)\/[^/]+\.ya?ml$ for manager github-actions (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|\/)action\.ya?ml$ for manager github-actions (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci-include (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)go.mod$ for manager gomod (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|\/)gradle\.properties$ for manager gradle (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|\/)gradle\/.+\.toml$ for manager gradle (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.versions\.toml$ for manager gradle (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper.properties$ for manager gradle-wrapper (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)requirements\.yaml$ for manager helm-requirements (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)values.yaml$ for manager helm-values (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)helmfile.yaml$ for manager helmfile (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Chart.yaml$ for manager helmv3 (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.html?$ for manager html (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)plugins\.(txt|ya?ml)$ for manager jenkins (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)kustomization\.yaml for manager kustomize (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.pom\.xml$ for manager maven (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)pom\.xml$ for manager maven (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)package.js$ for manager meteor (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/).node-version$ for manager nodenv (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)package.json$ for manager npm (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.(?:props|targets)$ for manager nuget (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|\/)dotnet-tools\.json$ for manager nuget (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|\/)global\.json$ for manager nuget (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.nvmrc$ for manager nvm (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)([\w-]*)requirements\.(txt|pip)$ for manager pip_requirements (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)setup.py$ for manager pip_setup (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Pipfile$ for manager pipenv (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.pre-commit-config\.yaml$ for manager pre-commit (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/).python-version$ for manager pyenv (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.sbt$ for manager sbt (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)setup\.cfg$ for manager setup-cfg (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)Package\.swift for manager swift (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: \.tf$ for manager terraform (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.terraform-version$ for manager terraform-version (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)terragrunt\.hcl$ for manager terragrunt (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: (^|/)\.terragrunt-version$ for manager terragrunt-version (repository=mn-dnp/playground/renovate)
DEBUG: Using file match: ^.travis.yml$ for manager travis (repository=mn-dnp/playground/renovate)
DEBUG: Matched 1 file(s) for manager gitlabci: .gitlab-ci.yml (repository=mn-dnp/playground/renovate)
DEBUG: Matched 1 file(s) for manager gitlabci-include: .gitlab-ci.yml (repository=mn-dnp/playground/renovate)
DEBUG: Matched 1 file(s) for manager pipenv: Pipfile (repository=mn-dnp/playground/renovate)
DEBUG: pipenv.extractPackageFile() (repository=mn-dnp/playground/renovate)
DEBUG: Found gitlabci package files (repository=mn-dnp/playground/renovate)
DEBUG: Found pipenv package files (repository=mn-dnp/playground/renovate)
DEBUG: Found 2 package file(s) (repository=mn-dnp/playground/renovate)
 INFO: Dependency extraction complete (repository=mn-dnp/playground/renovate)
       "baseBranch": "main",
       "stats": {
         "managers": {
           "gitlabci": {"fileCount": 1, "depCount": 1},
           "pipenv": {"fileCount": 1, "depCount": 12}
         },
         "total": {"fileCount": 2, "depCount": 13}
       }
DEBUG: baseBranch: main (repository=mn-dnp/playground/renovate)
DEBUG: Failed to look up dependency aws-cdk.aws-iam (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-iam)
DEBUG: Failed to look up dependency aws-cdk.aws-apigateway (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-apigateway)
DEBUG: Failed to look up dependency aws-cdk.aws-ecs (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-ecs)
DEBUG: Failed to look up dependency aws-cdk.aws-ecs-patterns (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-ecs-patterns)
DEBUG: Failed to look up dependency aws-cdk.aws-ec2 (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-ec2)
DEBUG: Failed to look up dependency aws-cdk.aws-secretsmanager (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-secretsmanager)
DEBUG: Failed to look up dependency aws-cdk.aws-route53 (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-route53)
DEBUG: Failed to look up dependency aws-cdk.aws-s3-assets (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-s3-assets)
DEBUG: Failed to look up dependency aws-cdk.aws-lambda (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-lambda)
DEBUG: Failed to look up dependency aws-cdk.aws-s3 (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-s3)
DEBUG: Failed to look up dependency aws-cdk.aws-ssm (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.aws-ssm)
DEBUG: Failed to look up dependency aws-cdk.core (repository=mn-dnp/playground/renovate, packageFile=Pipfile, dependency=aws-cdk.core)
DEBUG: Package releases lookups complete (repository=mn-dnp/playground/renovate)
       "baseBranch": "main"
DEBUG: branchifyUpgrades (repository=mn-dnp/playground/renovate)
DEBUG: 0 flattened updates found:  (repository=mn-dnp/playground/renovate)
DEBUG: Returning 0 branch(es) (repository=mn-dnp/playground/renovate)
DEBUG: config.repoIsOnboarded=true (repository=mn-dnp/playground/renovate)
DEBUG: packageFiles with updates (repository=mn-dnp/playground/renovate)
       "config": {
         "gitlabci": [
           {
             "packageFile": ".gitlab-ci.yml",
             "deps": [
               {
                 "skipReason": "contains-variable",
                 "replaceString": "${PARENT_IMAGE}",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "image",
                 "depIndex": 0,
                 "updates": []
               }
             ]
           }
         ],
         "pipenv": [
           {
             "packageFile": "Pipfile",
             "deps": [
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-apigateway",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-apigateway",
                     "message": "Failed to look up dependency aws-cdk.aws-apigateway"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-ec2",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-ec2",
                     "message": "Failed to look up dependency aws-cdk.aws-ec2"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-ecs",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 2,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-ecs",
                     "message": "Failed to look up dependency aws-cdk.aws-ecs"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-ecs-patterns",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 3,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-ecs-patterns",
                     "message": "Failed to look up dependency aws-cdk.aws-ecs-patterns"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-iam",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 4,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-iam",
                     "message": "Failed to look up dependency aws-cdk.aws-iam"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-lambda",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 5,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-lambda",
                     "message": "Failed to look up dependency aws-cdk.aws-lambda"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-route53",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 6,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-route53",
                     "message": "Failed to look up dependency aws-cdk.aws-route53"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-s3",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 7,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-s3",
                     "message": "Failed to look up dependency aws-cdk.aws-s3"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-s3-assets",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 8,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-s3-assets",
                     "message": "Failed to look up dependency aws-cdk.aws-s3-assets"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-secretsmanager",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 9,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-secretsmanager",
                     "message": "Failed to look up dependency aws-cdk.aws-secretsmanager"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.aws-ssm",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 10,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.aws-ssm",
                     "message": "Failed to look up dependency aws-cdk.aws-ssm"
                   }
                 ],
                 "versioning": "pep440"
               },
               {
                 "depType": "dev-packages",
                 "depName": "aws-cdk.core",
                 "managerData": {},
                 "currentValue": "==1.129.0",
                 "datasource": "pypi",
                 "depIndex": 11,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "aws-cdk.core",
                     "message": "Failed to look up dependency aws-cdk.core"
                   }
                 ],
                 "versioning": "pep440"
               }
             ],
             "registryUrls": ["https://pypi.org/simple"],
             "lockFiles": ["Pipfile.lock"],
             "constraints": {"python": "== 3.10.*"}
           }
         ]
       }
DEBUG: processRepo() (repository=mn-dnp/playground/renovate)
DEBUG: Processing 0 branches:  (repository=mn-dnp/playground/renovate)
DEBUG: Calculating hourly PRs remaining (repository=mn-dnp/playground/renovate)
DEBUG: currentHourStart=2021-11-16T08:00:00.000+00:00 (repository=mn-dnp/playground/renovate)
DEBUG: PR hourly limit remaining: 2 (repository=mn-dnp/playground/renovate)
DEBUG: Calculating prConcurrentLimit (5) (repository=mn-dnp/playground/renovate)
DEBUG: 0 PRs are currently open (repository=mn-dnp/playground/renovate)
DEBUG: PR concurrent limit remaining: 5 (repository=mn-dnp/playground/renovate)
DEBUG: Calculated maximum PRs remaining this run (repository=mn-dnp/playground/renovate)
       "prsRemaining": 2
DEBUG: PullRequests limit = 2 (repository=mn-dnp/playground/renovate)
DEBUG: Calculating hourly PRs remaining (repository=mn-dnp/playground/renovate)
DEBUG: currentHourStart=2021-11-16T08:00:00.000+00:00 (repository=mn-dnp/playground/renovate)
DEBUG: PR hourly limit remaining: 2 (repository=mn-dnp/playground/renovate)
DEBUG: Calculating branchConcurrentLimit (5) (repository=mn-dnp/playground/renovate)
DEBUG: 0 already existing branches found:  (repository=mn-dnp/playground/renovate)
DEBUG: Branch concurrent limit remaining: 5 (repository=mn-dnp/playground/renovate)
DEBUG: Calculated maximum branches remaining this run (repository=mn-dnp/playground/renovate)
       "branchesRemaining": 2
DEBUG: Branches limit = 2 (repository=mn-dnp/playground/renovate)
DEBUG: Ensuring Dependency Dashboard (repository=mn-dnp/playground/renovate)
DEBUG: ensureIssue() (repository=mn-dnp/playground/renovate)
DEBUG: Updating issue (repository=mn-dnp/playground/renovate)
DEBUG: Removing any stale branches (repository=mn-dnp/playground/renovate)
DEBUG: config.repoIsOnboarded=true (repository=mn-dnp/playground/renovate)
DEBUG: No renovate branches found (repository=mn-dnp/playground/renovate)
DEBUG: ensureIssueClosing() (repository=mn-dnp/playground/renovate)
DEBUG: Repository timing splits (milliseconds) (repository=mn-dnp/playground/renovate)
       "splits": {"init": 2522, "extract": 459, "lookup": 671, "update": 135},
       "total": 5088
DEBUG: http statistics (repository=mn-dnp/playground/renovate)
       "urls": {
         "[REDACTED]/api/v4/projects/mn-dnp%2Fplayground%2Frenovate (GET)": 1,
         "[REDACTED]/api/v4/projects/mn-dnp%2Fplayground%2Frenovate/issues (GET)": 1,
         "[REDACTED]/api/v4/projects/mn-dnp%2Fplayground%2Frenovate/issues/3 (GET)": 2,
         "[REDACTED]/api/v4/projects/mn-dnp%2Fplayground%2Frenovate/issues/3 (PUT)": 1,
         "[REDACTED]/api/v4/projects/mn-dnp%2Fplayground%2Frenovate/merge_requests (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-apigateway/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-ec2/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-ecs-patterns/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-ecs/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-iam/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-lambda/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-route53/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-s3-assets/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-s3/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-secretsmanager/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-aws-ssm/ (GET)": 1,
         "https://pypi.org/simple/aws-cdk-core/ (GET)": 1
       },
       "hostStats": {
         "[REDACTED]": {
           "requestCount": 6,
           "requestAvgMs": 358,
           "queueAvgMs": 0
         },
         "pypi.org": {"requestCount": 12, "requestAvgMs": 117, "queueAvgMs": 0}
       },
       "totalRequests": 18
 INFO: Repository finished (repository=mn-dnp/playground/renovate)
       "durationMs": 5088
DEBUG: Renovate exiting
```

</details>

### Have you created a minimal reproduction repository?

* https://gitlab.com/sandsoft/reproduction-repositories/renovate-12682
* https://github.com/tomirio619/renovate-12682

### PR DESCRIPTION
**Pull request by tgroemer**
## Changes:

Normalize package names and pypi simple link texts before extracting version numbers.

## Context:

According to PEP 426 package comparison must be case insensitive and hyphens and underscores must be considered equivalent.
The pull request allows using case insensitive package names when looking up packages on simple repositories. It also fixes problems with extracting versions for packages that contain periods. These were probably introduced with #12273 unintentionally.
See also #11139.

Closes #12682

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/pypi/index.spec.ts [hunk 1]
```diff
@@ -304,6 +307,23 @@ describe('datasource/pypi/index', () => {
...
+      const config = {
+        registryUrls: ['https://pypi.org/simple/'],
+      };
+      expect(
+        await getPkgReleases({
+          datasource,
+          ...config,
+          depName: 'package--with-hyphens',
+        })
+      ).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/pypi/index.spec.ts:324]: Please add partial response object here to have a more precise tests

File: lib/datasource/pypi/index.spec.ts [hunk 2]
```diff
@@ -304,6 +307,23 @@ describe('datasource/pypi/index', () => {
...
+      expect(httpMock.getTrace()).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/pypi/index.spec.ts:325]: Remove those, no longer needed.

**Review by viceice — COMMENTED** [review]
Needs some real tests

File: lib/datasource/pypi/index.spec.ts (hunk 1 above)
Comment (tgroemer) [reply lib/datasource/pypi/index.spec.ts]: Is it better now, did you mean something like that?

File: lib/datasource/pypi/index.ts [hunk 3]
```diff
@@ -164,27 +171,25 @@ export class PypiDatasource extends Datasource {
...
     // pep-0427 wheel packages
     //  {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl.
+    const wheelText = text.toLowerCase();
     const wheelPrefix = packageName.replace(regEx(/[^\w\d.]+/g), '_') + '-';
     const wheelSuffix = '.whl';
     if (
-      text.startsWith(wheelPrefix) &&
-      text.endsWith(wheelSuffix) &&
-      text.split('-').length > 2
+      wheelText.startsWith(wheelPrefix) &&
```
Comment (rarkins) [inline lib/datasource/pypi/index.ts:188]: Should wheelPrefix be lowercased too if it is to be compared to wheelText?

Comment (rarkins) [reply lib/datasource/pypi/index.ts:188]: I think the packageName here is already normalized

**Review by rarkins — APPROVED** [review]

**Pull request merged**